src/audio/module_adapter/module/modules.c

```diff
@@ -384,7 +405,11 @@ static int modules_reset(struct processing_module *mod)
 static const struct module_interface interface = {
 	.init = modules_init,
 	.prepare = modules_prepare,
-	.process_raw_data = modules_process,
+#if CONFIG_IPC_MAJOR_4
+	.process = modules_process,
+#else
+	.process_raw_data = modules_raw_process,
+#endif
```

Comment:
wouldn't https://github.com/thesofproject/sof/pull/8180/commits/28735353cbdaa52eaa0d2ad2ac8cf4a89dec3c28 be a better solution?

Reply:
please check this https://github.com/thesofproject/sof/pull/8554#pullrequestreview-1757520083.  So I didn't use it

Reply:
@RanderWang ok, we can use that work-around, but then you shouldn't need to touch modules.c ops at all - they're replaced by smart-amp's ops, aren't they?

Reply:
@lyakh  this modules.c is used by loadable module which use "struct module_interface interface" as ops and in this ops to invoke loadable module's ops, right ?

Build-in smart amp doesn't use this ops.

Reply:
if you use the work-around, proposed by @softwarecki in https://github.com/softwarecki/sof/commit/26f4b475d30c81477ad8b4eadf29abc48ab1e31e then these ops won't be used by the smart-amp test when built as a loadable module

Reply:
@lyakh  Actually I first used this patch and we have a discussion https://github.com/thesofproject/sof/pull/8580#discussion_r1417587494

Reply:
So we have 2 months until v2.9, lets refine this further as module support is still moving quickly. i.e. we can see how things work in practice with service API modules and Zephyr modules and see what solution for above supports both best.